crates/node-db-sql/sql/insert/block.sql

```diff
@@ -1,4 +1,29 @@
 INSERT
-    OR IGNORE INTO block (block_address, number, timestamp_secs, timestamp_nanos)
+    OR IGNORE INTO block (
+        block_address,
+        parent_block_id,
+        number,
+        timestamp_secs,
+        timestamp_nanos
+    )
 VALUES
-    (:block_address, :number, :timestamp_secs, :timestamp_nanos);
+    (
+        :block_address,
+        (
+            SELECT
+                COALESCE(
+                    (
+                        SELECT
+                            id
+                        FROM
+                            block
+                        WHERE
+                            block_address = :parent_block_address
+                    ),
+                    0 -- Default to for big bang block to it's own parent
```

Comment:
Yeh I wasn't totally happy with this but I think it's more of a problem with how we handle out of order blocks. If we made block parent nullable you'd have the same problem, you'd get None instead of 0. Maybe in the db it could be nullable but not in the type.

We could store the parent hash but I guess then we double the amount of hashes in the db.

I think we might have to have a table like `orphaned` that lists the ids of blocks with their expected parent hash then when the parent block comes in we replace null with the parent id or something.

As for query yeh you really can't query an orphaned branch. Maybe we can't consider blocks finalized until they aren't in the orphaned table their parent is finalized. That way you'd only have to get back to the parent instead of the BBB? 

The other option would be to just have a completely separate table `pending_blocks` to insert out of order blocks while they wait for their parent. This might make more sense as they should really be kept out of the system. You don't want to validate a block without a parent as that's a DOS vector. I'm leaning more towards this option now that I think about it so then you could consider all blocks in the actual `block` table to have parents.

Reply:
https://github.com/essential-contributions/essential-node/issues/155